lib/broadway_rabbitmq/producer.ex

```diff
@@ -122,6 +122,11 @@ defmodule BroadwayRabbitMQ.Producer do
   method, however, it removes the ability of the GenStage producer to control
   the demand. Therefore we need to use the `:prefetch_count` option to
   impose back-pressure at the channel level.
+  
+  Note that `:prefetch_count` [limits the number of *unacknowledged*
+  messages](https://www.rabbitmq.com/consumer-prefetch.html).
+  This count should be *at least* as large as the number of processors,
+  otherwise not all processors will receive messages.
```

Comment:
I think the math is that we want it to be max_demand * number_of_processors for full utilization. You can reduce the count or number of producers or max_demand if you want a lower flow.